src/node_alive/autostarter.py

```diff
@@ -0,0 +1,142 @@
+
+        while self._pending_resp is None and not rospy.is_shutdown():
+            rate.sleep()
+            # if there is no response after getting a request. It will set the request to None again and waits for new
+            # new request from service.
+            if rospy.Time.now() > t_end:
+                self._pending_req = None
+                self._pending_resp = AutoStarterCommandResponse(AutoStarterCommandResponse.LAUNCH_ERROR)
+                rospy.logwarn("No response after getting request. No launch files launched. "
+                              "Request is set to None and waits for new request from service")
+                break
+
+        response = self._pending_resp
+        self._pending_resp = None
+        return response
+
+    def _handle_auto_starter_command(self, req):
+        """
+        All request from the service are passed into this function and depending on the values it will either shutdown,
+        restart the launch file or start another launch file.
+        :param req: (AutoStarterCommandResponse) the command request that the service receives from the client contains
+        two data types, which launch file and which command.
+        :return AutoStarterCommandResponse:
+        """
+        if req.filename in self._launch_files_list:
+            if req.command == AutoStarterCommandRequest.START:
+                if self._launch is None:
+                    self.start(self._bringup_package_path , os.path.join("launch", "state_machines", req.filename))
+                    rospy.loginfo("Performed new start")
+                    return AutoStarterCommandResponse(AutoStarterCommandResponse.SUCCEEDED)
+                else:
+                    self.stop()
+                    self.start( self._bringup_package_path, os.path.join("launch", "state_machines", req.filename))
+                    rospy.loginfo("Performed restart")
+                    return AutoStarterCommandResponse(AutoStarterCommandResponse.SUCCEEDED)
+            elif req.command == AutoStarterCommandRequest.STOP:
+                rospy.loginfo("Performing shutdown")
+                self.stop()
+                rospy.loginfo("Performed shutdown")
+                return AutoStarterCommandResponse(AutoStarterCommandResponse.SUCCEEDED)
+            else:
+                return AutoStarterCommandResponse(AutoStarterCommandResponse.LAUNCH_ERROR)
+        else:
+            return AutoStarterCommandResponse(AutoStarterCommandResponse.FILE_NOT_PRESENT)
+
+    def start(self, package, path):
+        """
+        Before a launche file is started, it kills the node state_machine to make sure that all running launch files
+        are closed. Then it finds the right directory of the launch file and launches the file.
+        :param package: package of the current launch file
+        :param path: path to the current launch file
+        """
+        rosnode.kill_nodes(['/state_machine'])
+        rospy.sleep(0.1)
```

Comment:
Does accept rospy.sleep accept floats? I doubt it. If you don't want to wait for whole seconds, you probably need to use a different argument.